modules/packages/ZMQ.chpl

```diff
@@ -847,14 +847,13 @@ module ZMQ {
 
     // recv, strings
     pragma "no doc"
-    proc recv(type T, flags: int = 0) where isString(T) {
+    proc recv(type T, flags: int = 0) throws where isString(T) {
       var ret: T;
       on classRef.home {
         // Initialize an empty ZeroMQ message
         var msg: zmq_msg_t;
         if (0 != zmq_msg_init(msg)) {
-          var errmsg = zmq_strerror(errno):string;
-          try! halt("Error in Socket.recv(%s): %s\n".format(string:string, errmsg));
+          try throw_socket_error("Error in Socket.recv(%s): %s\n");
```

Comment:
I don't like the way that throw_socket_error assumes that the string has 2 format specifiers in it.
Can we just make throw_socket_error take in "send" or "recv" and put together the rest of the existing error format?